os/arch/arm/src/amebad/amebad_flash.c

```diff
@@ -159,35 +152,12 @@ static ssize_t amebad_flash_write(size_t addr, const void *buf, size_t length)
     flash_t flash;
     int32_t result = 0;
     ssize_t ret = 0;
-    irqstate_t irqs;
-
-    /* Disable IRQs while erasing sector */
-    irqs = irqsave();
-    if ((addr & 0xff) != 0) {
-        //! value should be 256 byte alignment
-        ret = -EFAULT;
-    } else if ((length & 0x3) != 0) {
-        ret = -EINVAL;
+    result = flash_stream_write(&flash, addr, length, buf);
+    if (result < 0) {
+        ret = -EIO;

```

Comment:
ret is not necessary actually, you can just return -EIO or length.
But flash_stream_write should check how many byte being written, and  each of error type(ex. if length is smaller than 0, return -EINVAL)

Reply:
Sure, that will be modified. Thanks!